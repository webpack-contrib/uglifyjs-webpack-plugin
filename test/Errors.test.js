/* eslint-disable
  import/order
 */
import webpack from './helpers/compiler';
import { cleanErrorStack } from './helpers';
import UglifyJsPlugin from '../src/index';

describe('Errors', () => {
  test('ValidationError', async () => {
    const config = {
      plugins: [
        new UglifyJsPlugin({
          uglifyOptions: {
            output: {
              invalid: true,
            },
          },
        }),
      ],
    };

    const stats = await webpack('entry.js', config);
    const { assets } = stats.compilation;

    const errors = stats.compilation.errors.map(cleanErrorStack);
    const warnings = stats.compilation.warnings.map(cleanErrorStack);

    expect(errors).toMatchSnapshot();
    expect(warnings).toMatchSnapshot();

    expect(assets['main.bundle.js'].source()).toMatchSnapshot();
  });

  test('Uglify Errors', async () => {
    const config = {
      plugins: [
        new UglifyJsPlugin({
          uglifyOptions: {
            output: {
              invalid: true,
            },
          },
        }),
      ],
    };

    const stats = await webpack('entry.js', config);
    const { errors } = stats.compilation;

    expect(errors[0].message).toEqual(expect.stringContaining('from UglifyJs'));
  });
});
