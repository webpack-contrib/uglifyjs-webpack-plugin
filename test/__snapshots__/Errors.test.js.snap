// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors ValidationError 1`] = `
Array [
  "Error: main.bundle.js from UglifyJs
DefaultsError: \`invalid\` is not a supported option",
  "Error: runtime.bundle.js from UglifyJs
DefaultsError: \`invalid\` is not a supported option",
]
`;

exports[`Errors ValidationError 2`] = `
Array [
  "NoModeWarning: configuration
The 'mode' option has not been set. Set 'mode' option to 'development' or 'production' to enable defaults for this enviroment. ",
]
`;

exports[`Errors ValidationError 3`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

// foo
/* @preserve*/
// bar
const a = 2 + 2;

module.exports = function Foo() {
  const b = 2 + 2;
  console.log(b + 1 + 2);
};


/***/ })
],[[0,1,0]]]);
//# sourceMappingURL=main.bundle.js.map"
`;
